sale_blanket_order/models/blanket_orders.py

```diff
@@ -264,9 +264,24 @@ def onchange_partner_id(self):
             values["team_id"] = self.partner_id.team_id.id
         self.update(values)
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        translated_draft = _("Draft")
+        for vals in vals_list:
+            if (
+                self.env.company.blanket_order_seq_number_from_draft
```

Comment:
**issue**: this `self.env.company` should be `vals["company_id"]` (possibly with `self.env.company` as a fallback)

My active company could be company A, but I could still be creating an order for company B that has a different setting. The BO needs to follow the setting specified in `company_id` regardless of what `self.env.company` is.